src/opm/parser/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -55,6 +57,41 @@
 
 #include <opm/parser/eclipse/EclipseState/Grid/EclipseGrid.hpp>
 
+namespace {
+
+#ifdef HAVE_OPENMP
+#include <omp.h>
+
+class ThreadGuard {
+public:
+    ThreadGuard(int num_threads) :
+        org_threads( omp_get_num_threads() )
```

Comment:
I do neither particularly like OpenMP nor do I have extended knowledge (hence I should probably better shut up), but:

The documentation say that this function return the number of threads in a parallel region, but the call is outside of any parallel region. Hence it is unclear what it returns. My assumption would be 1. Which would mean that the call of opm_set_num_threads might deactivate any later OpenMP parallelism.

Reply:
Hmmm - I had understood it differently, but you might be right--will have to check again.